src/groupeddataframe/splitapplycombine.jl

```diff
@@ -499,25 +512,40 @@ wrap(x::AbstractMatrix) =
     NamedTuple{Tuple(gennames(size(x, 2)))}(Tuple(view(x, :, i) for i in 1:size(x, 2)))
 wrap(x::Any) = (x1=x,)
 
-wrap_table(x::Any) =
+wrap_table(x::Any, ::Val) =
     throw(ArgumentError("return value must not change its kind " *
                         "(single row or variable number of rows) across groups"))
-wrap_table(x::Union{NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}},
-                    AbstractDataFrame,
-                    AbstractVector,
-                    AbstractMatrix}) = wrap(x)
+function wrap_table(x::Union{NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}},
+                             AbstractDataFrame, AbstractMatrix}, ::Val{T}) where T
+    T || throw(ArgumentError("mixing single and multiple column values is not allowed"))
```

Comment:
Maybe this would be easier to understand?
```suggestion
    T || throw(ArgumentError("function must return only single-column values, or only multiple-column values"))
```

Reply:
OK - I will add it locally due to conflicting changes I have locally.